runtime/protocol/http/common/test/aws/smithy/kotlin/runtime/http/UrlTest.kt

```diff
@@ -168,10 +201,60 @@ class UrlTest {
         }
     }
 
+    @Test
+    fun itParsesMinimum() =
```

Comment:
If we are set on rolling our own URL parsing then we likely need a much more expansive test suite.

For reference here is okhttp's URL parsing tests: https://github.com/square/okhttp/blob/master/okhttp/src/jvmTest/java/okhttp3/HttpUrlTest.kt

We were relying on the JVM stdlib before so we could get away with a small set of sanity tests. Rolling our own may be the right thing to do long term for KMP but it's also something we may want to punt on until we need to pay the cost. 

My 2 cents would be to use the stdlib for now still if we can and lump this into the cost of delivering KMP rather than the cost of delivering endpoints 2.0